### PR TITLE
support aten::type_as converter

### PR DESCRIPTION
**Pull request by inocsin**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)
support aten::type_as(Tensor self, Tensor other) -> (Tensor)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/type_as.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include "torch/torch.h"
+
+namespace trtorch {
+namespace core {
+namespace conversion {
+namespace converters {
+namespace impl {
+namespace {
+
+auto expand_registrations TRTORCH_UNUSED = RegisterNodeConversionPatterns().pattern(
```
Comment (peri044) [inline core/conversion/converters/impl/type_as.cpp:14]: expand_registrations -> type_as_registrations ?

**Review by peri044 — CHANGES_REQUESTED** [review]
Can you provide an example of the pytorch op in python which generates `aten::type_as` operation if you have ?

**Comment by inocsin** [comment]
> Can you provide an example of the pytorch op in python which generates `aten::type_as` operation if you have ?
```
  %138 : Tensor = aten::ne(%input_ids.1, %5)
  %139 : Tensor = aten::cumsum(%138, %5, %17) 
  %140 : Tensor = aten::type_as(%139, %138)
  %incremental_indices.1 : Tensor = aten::mul(%140, %138)
  %142 : Tensor = aten::add(%incremental_indices.1, %20, %5)
  %inputs_embeds.1 : Tensor = aten::embedding(%21, %input_ids.1, %5, %16, %16)
```
convert int to bool

**Comment by peri044** [comment]
This can be lowered to aten::to.other for which there is an existing PR https://github.com/NVIDIA/TRTorch/pull/382
There's more functionality that we plan to add to casting which is why it is in progress and will be worked on next week.

**Comment by peri044** [comment]
Closing this in favor of a lowering pass implementation https://github.com/NVIDIA/TRTorch/pull/382

**Pull request closed without merging**